Fix: issue #9 - Components in React

victorvictoria-maker opened the pull request.

victorvictoria-maker committed:
Fixed issue #9 by working on React components as part of the note.

Stan015 commented on pages/component.md:
```diff
@@ -8,14 +8,402 @@ hideInToc: true
+  <img src="/images/house.jpg" alt="My Image"  />
+</div>
+<div>
+<p v-click>A house is made up of reusable elements: <u class="text-teal-400">doors</u>, <u class="text-teal-400">windows</u>, <u class="text-teal-400">rooms</u>, <u class="text-teal-400">furniture</u>, <u class="text-teal-400">lights</u>, and more. You can change one element (like the lights) without affecting others. These are the components of a house. Similarly, React components allow you to d ...
+
+<p v-click>To build the house, you break it down into smaller, manageable parts, just as <u class="text-teal-400">React breaks a user interface into components</u>. Each part has a specific role, and all these parts work together to form a functional, maintainable whole. </p>
+
+</div>
+
+</div>
+<p v-click>COMPONENTS ARE THE FOUNDATION UPON WHICH USER INTERFACES (UI) ARE BUILT.</p>
+<div v-click class="flex gap-3 text-justify mt-6">
+<div>
+
+In React, a website can be broken into smaller components like a <u class="text-teal-400">header</u>, <u class="text-teal-400">sidebar</u>, <u class="text-teal-400">footer</u>, <u class="text-teal-400">main content</u>, and <u class="text-teal-400">buttons</u>. Each component combines its own markup, CSS, and JavaScript, making it unique, reusable, and easy to manage. Together, these components f ...
```
For better UX, I suggest you remove all underlines from highlighted words and sentences since they are not links. A reader might be trying to click them to read more just as I tried to. 

Do not remove the highlight colour, just the underlines as they are much. The highlight colour is okay.

Example fix: [doors]{.text-teal-400}

This should make the door highlighted in teal colour but not underlined...same fix for all underlines in this page. I am not suggesting you stop using underline, but underline is best for leading sentence or word in list items or links if any (e,g. the underline here in the following slide is okay; "Define the component function:")

victorvictoria-maker replied:
This has been adjusted as suggested. Thank you.

Stan015 commented on pages/component.md:
````diff
@@ -8,14 +8,402 @@ hideInToc: true
+    </div>
+  )
+}
+```
+
+</div>
+
+<div v-click> 
+2️⃣ Destructure Props: Extract props directly in the function parameters, with the option to assign default values.
+
+```jsx
+function Avatar({ person, size = 100 }) {
+  return (
+    <img
+      src={`https://i.imgur.com/${person.imageId}.jpg`}
+      alt={person.name}
+      width={size}
+      height={size}
+    />
+  )
+}
+```
+
+</div>
+</div>
+
+<div v-click>
+Forwarding Props with Spread Syntax: You can forward all props to child components with ...props
+
+```jsx
+function Profile() {
+  return (
+    <div>
+      <Avatar {...props} />
+    </div>
+  )
+}
+```
+
+</div>
````
The content on the right hand side of the slide are partly cut off on my screen, same for the bottom div. Consider restructuring the entire slide or moving some items to another slide. Although reducing the div sizes can also fix the overflows while maintaining the same style.

victorvictoria-maker replied:
Thank you. I adjusted the codes already. Please check. Thank you.

Stan015 reviewed (commented):
Great work right here @victorvictoria-maker. 
My comments here are suggestions. The overall pr looks good to me.

victorvictoria-maker committed:
Made pr corrections

Oluwasetemi committed:
Refactor CSS classes and add new styles

- Added 'logo' class for React logos with hover animation
- Added 'btn' class for styled buttons
- Added 'card' class for styled cards
- Added new rules for 'w-fill', 'custom-shadow', and 'grid-cols-*'

Oluwasetemi commented:
@victoriavictor-maker I updated your PR with some little changes
